modules/subschema/src/main/scala/com.snowplowanalytics/iglu.schemaddl/jsonschema/subschema/package.scala

```diff
@@ -221,4 +251,17 @@ package object subschema {
     minCheck && maxCheck
   }
 
+  // Semigroup for Compatibility
+  def combine(c1: Compatibility, c2: Compatibility): Compatibility =
+    (c1, c2) match {
+      case (Compatible, Compatible) => Compatible
+      case (Incompatible, _) | (_, Incompatible) => Incompatible
+      case _ => Undecidable
+    }
+
+  // Emulates a NonEmptyList without relying on cats. Required because Compatibility cannot form a Monoid since
+  // an "empty" element doesn't exist
```

Comment:
Just a though, maybe we can consider the empty element, the Undecidable so:
```
Foldable[List].fold(List(Compatible, Incompatible)) // Incompatible
Foldable[List].fold(List.empty[Compatible]) // Undecidable
```

Reply:
I don't think that would work, as it would violate the law that `empty` [must be an identity for combine](https://typelevel.org/cats/typeclasses/monoid.html):
`combine(x, empty) = combine(empty, x) = x` would be false for `x = Compatible` and `empty = Undecidable`.
In practice, if you had a list full of `Compatible`, but you folded using `Undecidable`, you would start with `Undecidable` as the `zero` element and then you would get back `Undecidable` at the end (every op would be `combine(Undecidable, Compatible)`), even if all items in the list are actually == `Compatible`

Reply:
yeah exactly, so I thought I would just implement the combine and use reduce on a list we know is going to have at least one element. After all from both places where I invoked this we are guaranteed there is at least one item in the list (at least one patternProperty representing additionalItems for objects, and same for arrays, at least one item representing additionalItems)